src/discovery/Discovery.ts

```diff
@@ -1,262 +1,411 @@
+            // If the vertex we've found is our own node, we simply get our own chain
+            const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId)!;
+            if (nodeId.equals(this.nodeManager.getNodeId())) {
+              const vertexChainDataEncoded =
+                await this.nodeManager.getChainData();
+              // Decode all our claims - no need to verify (on our own sigchain)
+              for (const c in vertexChainDataEncoded) {
+                const claimId = c as ClaimIdEncoded;
+                vertexChainData[claimId] = claimsUtils.decodeClaim(
+                  vertexChainDataEncoded[claimId],
+                );
+              }
+              // Otherwise, request the verified chain data from the node
+            } else {
+              vertexChainData = await this.nodeManager.requestChainData(nodeId);
+            }
+            // TODO: for now, the chain data is treated as a 'disjoint' set of
+            // cryptolink claims from a node to another node/identity
+            // That is, we have no notion of revocations, or multiple claims to the
+            // same node/identity. Thus, we simply iterate over this chain of
+            // cryptolinks.
+            // Now have the NodeInfo of this vertex
+            const vertexNodeInfo: NodeInfo = {
+              id: nodesUtils.encodeNodeId(nodeId),
+              chain: vertexChainData,
             };
-            await this.gestaltGraph.linkNodeAndNode(
-              vertexNodeInfo,
-              linkedVertexNodeInfo,
-            );
-
-            // Add this vertex to the queue if it hasn't already been visited
-            const linkedVertexGK =
-              gestaltsUtils.keyFromNode(linkedVertexNodeId);
-            if (!visitedVertices.has(linkedVertexGK)) {
-              vertexQueue.push(linkedVertexGK);
+            // Iterate over each of the claims in the chain (already verified)
+            // TODO: because we're iterating over keys in a record, I don't believe
+            // that this will iterate in lexicographical order of keys. For now,
+            // this doesn't matter though (because of the previous comment).
+            for (const claimId in vertexChainData) {
+              const claim: Claim = vertexChainData[claimId as ClaimIdEncoded];
+              // If the claim is to a node
+              if (claim.payload.data.type === 'node') {
+                // Get the chain data of the linked node
+                // Could be node1 or node2 in the claim so get the one that's
+                // not equal to nodeId from above
+                const node1Id = nodesUtils.decodeNodeId(
+                  claim.payload.data.node1,
+                )!;
+                const node2Id = nodesUtils.decodeNodeId(
+                  claim.payload.data.node2,
+                )!;
+                const linkedVertexNodeId = node1Id.equals(nodeId)
+                  ? node2Id
+                  : node1Id;
```

Comment:
A quick note about this too. For all current node to node claim set-ups, I've set it such that `node1` refers to the node that the claim is appearing on (e.g. a claim from `A` to `B` on `A`'s sigchain will have `node1` set as `A`.

Obviously, this is a more robust solution though.

In the future, this could possibly be improved by giving more descriptive field names in the claim (e.g. `sourceNode`, `targetNode`, or some other variant).

Reply:
Ahh, yes I made this change when I was getting a weird bug where I was seeing node claims from nodes to themselves appearing in the gestalt graph (e.g. Node A claiming Node A). I thought that this is where the issue was coming from at first, however in the end I think the error was coming solely from the gestalt graph (https://github.com/MatrixAI/js-polykey/pull/320#issuecomment-1026394383). So this change can either remain or be reverted.

Reply:
That's fine - this can remain. I've made a comment about this here https://github.com/MatrixAI/js-polykey/issues/189#issuecomment-1038464478. Resolving this.